Dialog/TilesetEditDialog.cpp

```diff
@@ -1099,6 +1107,44 @@ void TilesetEditDialog::on_pushButton_ExportPalette_clicked()
         }
         delete[] palettedata;
     }
+    else if(selectedfilter.compare("Raw Binary palette (*.bin)") == 0)
+    {
+            unsigned short *palettedata = new unsigned short[16];
+            for(int j = 0; j < 16; ++j)
+            {
+                int red = (tmppalette[j] & 0xFF0000) >> 16; // R
+                int green = (tmppalette[j] & 0xFF00) >> 8; // G
+                int blue = tmppalette[j] & 0xFF; // B
+
+                //Going from 8 bits to 5 bits
+                red >>=3;
+                green >>=3;
+                blue >>=3;
+
+                //Assemble color from left to right with OR operator (blue->green->red)
+                short newcolor=0;
+                newcolor|=blue;
+                newcolor <<=5;
+                newcolor|=green;
+                newcolor <<=5;
+                newcolor|=red;
+                palettedata[j]=newcolor;
+            }
+            QFile palfile(qFilePath);
+            palfile.open(QIODevice::WriteOnly);
+            if (palfile.isOpen())
+            {
+                QDataStream out(&palfile);
+                out.setByteOrder(QDataStream::LittleEndian); // *** set little endian byte order
+                for (int i=0 ; i<16; i++) {
+                    out<<quint16(palettedata[i]);
+                }
+                palfile.close();
+            } else {
+                QMessageBox::critical(this, QString("Error"), QString("Cannot save file!"));
```

Comment:
Let's also include the error reason: https://doc.qt.io/qt-5/qfiledevice.html#error

Reply:
Ok I used https://doc.qt.io/qt-5/qiodevice.html#errorString and it seemed to work